tests/python_package_test/test_basic.py

```diff
@@ -913,6 +913,17 @@ def test_feature_names_are_set_correctly_when_no_feature_names_passed_into_Datas
     assert ds.construct().feature_name == ["Column_0", "Column_1", "Column_2"]
 
 
+@pytest.mark.parametrize("max_depth", [4, 8, 12, 16])
+def test_max_depth_is_enforced(capsys, max_depth):
+    X, y = make_blobs(n_samples=1_000, n_features=1, centers=2)
+    model = lgb.LGBMRegressor(objective="binary", max_depth=max_depth, device=getenv("TASK"), verbose=0)
```

Comment:
```suggestion
    model = lgb.LGBMRegressor(objective="binary", max_depth=max_depth, verbose=0)
```

Let's not do this, please.

1. these tests are often run outside of CI environments, where environment variable `TASK` isn't guaranteed to be set (what would happen if `device=None`?)
2. for CI, we use a different mechanism to ensure all tests run with `device="cuda"`:

https://github.com/microsoft/LightGBM/blob/a64e00b09e56ca994766f8eeb59845e49a0217fe/.ci/test.sh#L189-L193

Maybe that mechanism isn't great (replicating it locally can be a little annoying), but as long as it exists, the test should follow it as all other tests here do.